src/cascadia/TerminalSettingsModel/TerminalSettings.idl

```diff
@@ -16,6 +18,13 @@ namespace TerminalApp
                                     Microsoft.Terminal.TerminalControl.IControlSettings
     {
         TerminalSettings();
+        TerminalSettings(CascadiaSettings appSettings, Guid profileGuid, Microsoft.Terminal.TerminalControl.IKeyBindings keybindings);
+
+        static TerminalSettings BuildSettings(CascadiaSettings appSettings, NewTerminalArgs newTerminalArgs, Microsoft.Terminal.TerminalControl.IKeyBindings keybindings);

```

Comment:
I'd suggest just making this two static functions, `CreateWithProfileByID` and `CreateWithNewTerminalArgs`. Names are much clearer than constructors!

Reply:
For the record, I _really_ liked this idea :)